VimeoNetworking/Sources/Models/VIMUpload.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  VIMUpload.swift
+//  Pods
+//
+//  Created by Lehrer, Nicole on 4/18/18.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// Encapsulates the upload-related information on a video object.
+public class VIMUpload: VIMModelObject
+{
+    /// The approach for uploading the video, expressed as a Swift-only enum
+    ///
+    /// - streaming: Two-step upload without using tus; this will be deprecated
+    /// - post: Upload with an HTML form or POST
+    /// - pull: Upload from a video file that already exists on the internet
+    /// - tus: Upload using the open-source tus protocol
+    public enum UploadApproach: String
+    {
+        case streaming
+        case post
+        case pull
+        case tus
+    }
+    
+    /// The status code for the availability of the uploaded video, expressed as a Swift-only enum
+    ///
+    /// - complete: The upload is complete
+    /// - error: The upload ended with an error
+    /// - underway: The upload is in progress
+    public enum UploadStatus: String
+    {
+        case complete
+        case error
+        case underway = "in_progress"
+    }
+    
+    /// The approach for uploading the video
+    @objc dynamic public private(set) var approach: String?
+
+    /// The file size in bytes of the uploaded video
+    @objc dynamic public private(set) var size: NSNumber?
+
+    /// The status code for the availability of the uploaded video
+    @objc dynamic public private(set) var status: String?
+    
+    /// The HTML form for uploading a video through the post approach
+    @objc dynamic public private(set) var form: String?
+    
+    /// The link of the video to capture through the pull approach
+    @objc dynamic public private(set) var link: String?
+
+    /// The URI for completing the upload
+    @objc dynamic public private(set) var completeUri: String?
```

Comment:
I can't remember if we usually do `completeURI` instead.

Reply:
It looks like originally we did leave it lower case: https://github.com/vimeo/VimeoUpload/blob/develop/Pods/VimeoNetworking/VimeoNetworking/Sources/Models/VIMUploadTicket.h but I do prefer an all caps or all lowercase approach for acronyms. It's just a little more code - we convert `complete_uri` automatically to `completeUri` whereas now we need to use `getObjectMapping`.